File: Lightened Dream/AboutForm.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Linq;
using System.Windows.Forms;
using System.Net;
using System.IO;
using System.Threading;
namespace LightenedDream
{

	public partial class AboutForm
	{

		private string m_strUpdateURL = "http://www.lucid-code.com/LightenedDream";

		private bool m_boolClosing = false;
		private void AboutForm_Load(System.Object sender, System.EventArgs e)
		{
			lblVersion.Text = Application.ProductVersion;
		}

		private void tmrLoad_Tick(System.Object sender, System.EventArgs e)
		{
			tmrLoad.Enabled = false;


			try {
				UpdateClass objUpdater = new UpdateClass();
				Thread objUpdateThread = new Thread(objUpdater.CheckForUpdate);
				objUpdateThread.Start();

				do {
					Application.DoEvents();
					System.Threading.Thread.Sleep(10);

					if (objUpdater.Complete) {
						break; // TODO: might not be correct. Was : Exit Do
					}
					if (m_boolClosing) {
						//objUpdateThread.Abort()
						return;
					}
				} while (true);

				if (UpdateAvailable(objUpdater.Version)) {
					m_strUpdateURL = objUpdater.URL;
					lblUpdate.Visible = true;
					lblUpdate.Text = "A new version (" + objUpdater.Version + ") is available. Download now!";
				}

			} catch (Exception ex) {
			}
		}

		private bool UpdateAvailable(string NewVersion)
		{

			try {
				string OldVersion = lblVersion.Text;
				string[] arrOldVersion = Strings.Split(lblVersion.Text, ".");
				// 2.9.9.0
				string[] arrNewVersion = Strings.Split(NewVersion, ".");
				//    ' 2.9.9.1

				if (Conversion.Val(arrNewVersion[0]) > Conversion.Val(arrOldVersion[0])) {
					return true;
				} else {
					if (Conversion.Val(arrNewVersion[1]) > Conversion.Val(arrOldVersion[1])) {
						return true;
					} else {
						if (Conversion.Val(arrNewVersion[2]) > Conversion.Val(arrOldVersion[2])) {
							return true;
						} else {
							if (Conversion.Val(arrNewVersion[3]) > Conversion.Val(arrOldVersion[3])) {
								return true;
							} else {
								return false;
							}
						}
					}
				}
			} catch (Exception ex) {
				return false;
			}
		}

		private void lblUpdate_LinkClicked(System.Object sender, System.Windows.Forms.LinkLabelLinkClickedEventArgs e)
		{
			try {
				System.Diagnostics.Process.Start(m_strUpdateURL);
			} catch (Exception ex) {
				MessageBox.Show(ex.Message, "LightenedDream.About.Link()", MessageBoxButtons.OK, MessageBoxIcon.Error);
			}
		}

		private void LinkLabel1_LinkClicked(System.Object sender, System.Windows.Forms.LinkLabelLinkClickedEventArgs e)
		{
			try {
				System.Diagnostics.Process.Start("http://www.lucidcode.com/LightenedDream/");
			} catch (Exception ex) {
				MessageBox.Show(ex.Message, "LightenedDream.About.Link()", MessageBoxButtons.OK, MessageBoxIcon.Error);
			}
		}

		private void AboutForm_FormClosing(System.Object sender, System.Windows.Forms.FormClosingEventArgs e)
		{
			m_boolClosing = true;
		}

		private class UpdateClass
		{

			public string Version;
			public string URL;

			public bool Complete = false;
			public void CheckForUpdate()
			{

				try {
					WebRequest.DefaultWebProxy.Credentials = CredentialCache.DefaultCredentials;
					HttpWebRequest objRequest = (HttpWebRequest)WebRequest.Create("http://www.lucid-code.com/LightenedDream/updates/LatestVersion.html");
					objRequest.Method = "GET";
					HttpWebResponse objWebResponse = (HttpWebResponse)objRequest.GetResponse();
					StreamReader objResponseStream = new StreamReader(objWebResponse.GetResponseStream(), System.Text.Encoding.GetEncoding(1252));
					var strResponse = objResponseStream.ReadToEnd();
					objWebResponse.Close();
					objResponseStream.Close();

					if (strResponse.StartsWith("Version ")) {
						string[] arrLines = strResponse.Split(Constants.vbNewLine);
						Version = arrLines[0].Replace("Version ", "");
						URL = arrLines[1];
					}

				} catch (Exception ex) {
				}

				Complete = true;
			}

		}
	}
}

File: Lightened Dream/AboutForm.vb
Imports System.Net
Imports System.IO
Imports System.Threading

Public Class AboutForm

    Private m_strUpdateURL As String = "https://www.lucid-code.com/LightenedDream"
    Private m_boolClosing As Boolean = False

    Private Sub AboutForm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        lblVersion.Text = Application.ProductVersion
    End Sub

    Private Sub tmrLoad_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles tmrLoad.Tick
        tmrLoad.Enabled = False

        Try

            Dim objUpdater As New UpdateClass
            Dim objUpdateThread As New Thread(AddressOf objUpdater.CheckForUpdate)
            objUpdateThread.Start()

            Do
                Application.DoEvents()
                System.Threading.Thread.Sleep(10)

                If objUpdater.Complete Then
                    Exit Do
                End If
                If m_boolClosing Then
                    'objUpdateThread.Abort()
                    Exit Sub
                End If
            Loop

            If UpdateAvailable(objUpdater.Version) Then
                m_strUpdateURL = objUpdater.URL
                lblUpdate.Visible = True
                lblUpdate.Text = "A new version (" & objUpdater.Version & ") is available. Download now!"
            End If

        Catch ex As Exception
        End Try
    End Sub

    Private Function UpdateAvailable(ByVal NewVersion As String) As Boolean
        Try

            Dim OldVersion As String = lblVersion.Text
            Dim arrOldVersion() As String = Split(lblVersion.Text, ".") ' 2.9.9.0
            Dim arrNewVersion() As String = Split(NewVersion, ".") '    ' 2.9.9.1

            If Val(arrNewVersion(0)) > Val(arrOldVersion(0)) Then
                Return True
            Else
                If Val(arrNewVersion(1)) > Val(arrOldVersion(1)) Then
                    Return True
                Else
                    If Val(arrNewVersion(2)) > Val(arrOldVersion(2)) Then
                        Return True
                    Else
                        If Val(arrNewVersion(3)) > Val(arrOldVersion(3)) Then
                            Return True
                        Else
                            Return False
                        End If
                    End If
                End If
            End If
        Catch ex As Exception
            Return False
        End Try
    End Function

    Private Sub lblUpdate_LinkClicked(ByVal sender As System.Object, ByVal e As System.Windows.Forms.LinkLabelLinkClickedEventArgs) Handles lblUpdate.LinkClicked
        Try
            System.Diagnostics.Process.Start(m_strUpdateURL)
        Catch ex As Exception
            MessageBox.Show(ex.Message, "LightenedDream.About.Link()", MessageBoxButtons.OK, MessageBoxIcon.Error)
        End Try
    End Sub

    Private Sub LinkLabel1_LinkClicked(ByVal sender As System.Object, ByVal e As System.Windows.Forms.LinkLabelLinkClickedEventArgs) Handles lblWeb.LinkClicked
        Try
            System.Diagnostics.Process.Start("https://www.lucidcode.com/LightenedDream/")
        Catch ex As Exception
            MessageBox.Show(ex.Message, "LightenedDream.About.Link()", MessageBoxButtons.OK, MessageBoxIcon.Error)
        End Try
    End Sub

    Private Sub AboutForm_FormClosing(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosingEventArgs) Handles MyBase.FormClosing
        m_boolClosing = True
    End Sub

    Private Class UpdateClass

        Public Version As String
        Public URL As String
        Public Complete As Boolean = False

        Public Sub CheckForUpdate()
            Try

                WebRequest.DefaultWebProxy.Credentials = CredentialCache.DefaultCredentials
                Dim objRequest As HttpWebRequest = CType(WebRequest.Create("https://www.lucid-code.com/LightenedDream/updates/LatestVersion.html"), HttpWebRequest)
                objRequest.Method = "GET"
                Dim objWebResponse As HttpWebResponse = CType(objRequest.GetResponse(), HttpWebResponse)
                Dim objResponseStream As StreamReader = New StreamReader(objWebResponse.GetResponseStream(), System.Text.Encoding.GetEncoding(1252))
                Dim strResponse = objResponseStream.ReadToEnd()
                objWebResponse.Close()
                objResponseStream.Close()

                If strResponse.StartsWith("Version ") Then
                    Dim arrLines() As String = strResponse.Split(vbNewLine)
                    Version = arrLines(0).Replace("Version ", "")
                    URL = arrLines(1)
                End If

            Catch ex As Exception
            End Try

            Complete = True
        End Sub

    End Class
End Class